cmake/apply_lvi_mitigation.cmake

```diff
@@ -15,35 +15,62 @@ macro (get_glibc_version)
   endif ()
 endmacro ()
 
-function (apply_lvi_mitigation NAME)
+# Usage
+# apply_lvi_mitigation(
+#     <target>
+#     [OE_USE_CONTROLFLOW_GNU]
+# )
+#
+# OE_USE_CONTROLFLOW_GNU: Optional.
+# You should enable it if all of the below is true:
+# 1. You are the client, who builds app using installed OE
+# 2. The installed OE is built with ControlFlow-GNU
+# 3. You are building your app with ControlFlow-Clang
```

Comment:
Renamed to `OE_USED_CONTROLFLOW_GNU`

* It would be ideal if we avoid this and determine which one should be used without any additional work from users

I can only come up with 1 way to do that: generate a duplicate of `apply_lvi_mitigation.cmake` during `make install` so the OE build info can be fed into the code without the user manually assigning it. 

* From your table it seems like it is easier for users if OE is built with ControlFlow-Clang instead. Would you agree with that?

I think so. We changing LVI mitigation method won't affect user, if they are using ControlFlow-GNU.

Reply:
I think we can switch to build with ControlFlow-Clang by default without requiring user to make any change from their side.

Reply:
Sounds good. Let me look into that for the next release.

Reply:
Scratch that idea. We need to build with ControlFlow-GNU to ensure a complete mitigation.